### DISCUSSION TITLE
v1.4.x

### DISCUSSION THREAD

**Discussion by bjlittle** [SciTools/nc-time-axis#64] [Releases]
The release manager for `v1.4.x` is @spencerkclark 

The release feature branch is [v1.4.x](https://github.com/SciTools/nc-time-axis/tree/v1.4.x)

| Minor Release | Status | Milestone | Project | Date | RTD | Release | conda-forge | PyPI |
| :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: |
| `v1.4.0` | Completed | [v1.4.0](https://github.com/SciTools/nc-time-axis/milestone/2) | [v1.4.0](https://github.com/SciTools/nc-time-axis/projects/3) | 2021-10-23 | [Available](https://nc-time-axis.readthedocs.io/en/v1.4.0/) | [v1.4.0](https://github.com/SciTools/nc-time-axis/releases/tag/v1.4.0) | [Available](https://anaconda.org/conda-forge/nc-time-axis) | [Available](https://pypi.org/project/nc-time-axis/) |

| Patch Release | Status | Milestone | Project | Date | RTD | Release | conda-forge | PyPI |
| :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: |
| `v1.4.1` | Completed | [v1.4.1](https://github.com/SciTools/nc-time-axis/milestone/3) | None | 2022-04-20 | [Available](https://nc-time-axis.readthedocs.io/en/v1.4.1/) | [v1.4.1](https://github.com/SciTools/nc-time-axis/releases/tag/v1.4.1) | [Available](https://anaconda.org/conda-forge/nc-time-axis) | [Available](https://pypi.org/project/nc-time-axis/) |

**Comment by bjlittle** [comment]
Hey @spencerkclark, fancy taking on the release manager role for `1.4.0` ?

**Reply by spencerkclark** [reply]
Yes, I would be happy to take that on.  Thanks for managing `1.3.0`!

**Reply by bjlittle** [reply]
Thanks, that's just awesome @spencerkclark - you've totally just made my day :partying_face: 

Between `1.3.0` to `1.4.0` it would be great to bootstrap some minimal docs, which capture dev notes on the steps involved in a release - nothing heavy, just a light touch for guidance.

Ideally, it would be great to rotate the release manager (the task really isn't that onerous) so that we all get a change improve and shape the process for the better.

**Reply by bjlittle** [reply]
@spencerkclark Do you have a PyPI user?

If you're happy to share that with me then I'll add you as an owner to https://pypi.org/project/nc-time-axis/

Also, when we update the `nc-time-axis` recipe on `conda-forge` I'll add you the the list of recipe maintainers :+1:

**Reply by spencerkclark** [reply]
@bjlittle oops sorry I neglected to respond to this!  I do have a PyPI account.  My username is `spencerkclark` (the same as my GitHub username).

**Reply by spencerkclark** [reply]
> it would be great to bootstrap some minimal docs, which capture dev notes on the steps involved in a release - nothing heavy, just a light touch for guidance.

Indeed if you have any tips here that would be great.  I have released packages to PyPI before, but not `conda-forge`.  Would you suggest that [the release guide in the iris docs](https://scitools-iris.readthedocs.io/en/latest/developers_guide/release.html) is a decent guide for what needs to be done here?

**Reply by spencerkclark** [reply]
Gentle ping regarding PyPI @bjlittle -- I think once #87 is in we should be ready for the next release.

**Comment by spencerkclark** [comment]
I think we are getting close to being able to make a 1.4.0 release -- nearly all the tasks we initially tagged with the 1.4.0 milestone have been completed, as well as a few more.  While I posted a couple more issues (#85 and #88), I do not believe they are urgent.  

I propose we go forward once the following issues have been addressed, both of which I think we are close on:

- [x] #67 -- I left a comment there this morning.  My hunch is that we should continue to use `matplotlib.use("agg")` for the time being, but I'll leave the issue open for a couple days in case any others have opinions.
- [x] #81 -- I think we are close here too.  #86 goes through with adding a deprecation warning to the `CalendarDateTime` constructor, which will enable us to remove it in the 1.5 release of nc-time-axis.
- [x] #89 

Does that sound like a good plan?

**Comment by spencerkclark** [comment]
The v1.4.0 release is complete -- thanks for your help @aulemahal, @trexfeathers, and @bjlittle!  Everything seems to have gone smoothly.  It is available now on both [PyPI](https://pypi.org/project/nc-time-axis/) and [conda-forge](https://anaconda.org/conda-forge/nc-time-axis).
